File: Universidad de Granada/Programación y Diseño Orientada a Objetos/QytetetRuby/pkg/QytetetRuby-0.0.1/lib/casilla.rb
#encoding: utf-8
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

#authors: Miguel Ángel Benítez Alguacil
#         Najib Saadouli Arco

require_relative "tipo_casilla.rb"
require_relative "titulo_propiedad.rb"

module ModeloQytetet
  class Casilla
    #Constructor con todos los parámetros
    def initialize(numeroCasilla_, coste_, tipo_, titulo_)
      @numeroCasilla = numeroCasilla_
      @coste = coste_
      @tipo = tipo_
      @titulo = titulo_
    end
    
    #Constructor específico para las que no son tipo CALLE
    def self.crear_casilla(numeroCasilla_, tipo_, coste_)
      self.new(numeroCasilla_, coste_, tipo_, nil)
    end
    
    #Getters y Setter de los atributos
    attr_reader :numeroCasilla, :coste, :tipo
    attr_accessor :titulo
    
    #Otros métodos
    def soyEdificable
        false
    end
    
    def to_s
      "\n\tCasilla {numeroCasilla = #{@numeroCasilla}, coste = #{@coste}, tipo = #{@tipo}}"
    end
    
  end
end

File: Universidad de Granada/Programación y Diseño Orientada a Objetos/QytetetJava/src/modeloqytetet/Casilla.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package modeloqytetet;

/**
 *
 * @author Miguel Ángel Benítez Alguacil
 *         Najib Saadouli Arco
 */
public abstract class Casilla {
    private int numeroCasilla = 0;
    private int coste;

    public Casilla(int numeroCasilla, int coste) {
        this.numeroCasilla = numeroCasilla;
        this.coste = coste;
    }
    
    //Getters de los atributos de Casilla
    public int getNumeroCasilla() {
        return numeroCasilla;
    }

    public int getCoste() {
        return coste;
    }

    protected abstract TipoCasilla getTipo();

    public abstract TituloPropiedad getTitulo();
    
    public void setCoste(int coste) {
        this.coste = coste;
    }
    
    protected abstract boolean soyEdificable();
}
